nbconvert/nbconvertapp.py

```diff
@@ -30,6 +31,36 @@
 #Classes and functions
 #-----------------------------------------------------------------------------
 
+class CaselessStrEnumOrDottedObjectName(CaselessStrEnum, DottedObjectName):
```

Comment:
-1 on that because right now the `Enum` thing  for the error message if you give a wrong value:

```
...must be any of ['custom', 'html', 'latex', 'markdown', 'notebook', 'pdf', 'python', 'rst', 'script', 'slides'], but a value of 'tex' <class 'str'> was specified.
```

which goes away for unicode.